src/homework/MyFirstHomework.java

```diff
@@ -0,0 +1,107 @@
+package homework;
+
+public class MyFirstHomework {
+    public static void main(String[] args) {
+        //Easy peasy
+        int x = 10;
+        if (x > 0)
+            System.out.println("Positive number");
+
+        //Odd or even
+        int n = 16;
+        if (n%2==0)
+            System.out.println("Even number");
+
+        //Age group classifier
+        int age = 15;
+        if (age <= 17) {
+            System.out.println("Teenager");
+        }
+        else
+            System.out.println("Adult");
+
+        //Leap year checker
+        int year = 2024;
+        boolean leap = false;
+
+        if (year % 4 == 0) {
+            if (year % 100 == 0) {
+                if (year % 400 == 0)
+                    leap = true;
+                else
+                    leap = false;
+            }
+            else
+                leap = true;
+        }
+        else
+            leap = false;
+        if (leap)
+            System.out.println("Leap year");
```

Comment:
// Leap year is divisible by 4
        int year = 2024;
        boolean isLeapYear = false;
        if (year % 4 == 0) {
            // Leap year is either not divisible by 100 or divisible by 400
            if (year % 100 == 0) {
                if (year % 400 == 0) {
                    isLeapYear = true;
                }
            } else {
                isLeapYear = true;
            }
        }

        if (isLeapYear) {
            System.out.println(year + " is a leap year.");
        } else {
            System.out.println(year + " is not a leap year.");
        }
    }